### ISSUE TITLE
Empty tree should not be focusable

### ISSUE THREAD

**Issue by alexandrudima** [Microsoft/vscode#2347]
Testing #2145

e.g. debug variables tree

![image](https://cloud.githubusercontent.com/assets/5047891/12582422/54715330-c43d-11e5-9b5b-cb2b57218a0a.png)

**Comment by bpasero** [comment]
@alexandrudima an empty tree can still have a context menu associated. are you saying we should detect the case where the tree is empty && has no context menu? I am not sure this would confuse people why sometimes they see blue and sometimes they do not.

**Comment by alexandrudima** [comment]
Good point about the context menu case, so maybe it makes sense to tab into an empty explorer tree given that's the only place where a context menu can be invoked.

IMHO in this case, tabbing into an empty variable list in the debugger is pretty useless.

**Comment by bpasero** [comment]
@alexandrudima but that is up to the tree right? e.g. I can also tab into watch expressions to create a new one there.

**Comment by alexandrudima** [comment]
Right clicking in empty spaces appears to be common:
- e.g. in this webpage I can right click on the whitespace
- e.g. on my desktop I can right click anywhere on it
- however, the whitespace on this webpage is not "focusable" and there must be alternative ways to get those actions besides context menu...

**Comment by bpasero** [comment]
@alexandrudima sorry can you clarify what your last comment means?

**Comment by alexandrudima** [comment]
gif time :)

I can do a contextmenu with the mouse in areas where I cannot go to via tab focus. e.g. the whitespace on this webpage or the desktop. I only assume there must be alternative ways to trigger those actions since I cannot get to focus the whitespace to trigger a contextmenu with the keyboard

![contextmenu-even-in-nonfocusable](https://cloud.githubusercontent.com/assets/5047891/12583133/6e755a3e-c441-11e5-86f9-f97a051d3151.gif)

**Comment by bpasero** [comment]
Well yeah, might be. All I am saying is that imho it makes sense to draw the blue border in the tree as long as you can trigger the context menu (using the context menu key on the keyboard). Even if there are other ways I find this a nice gesture.

**Comment by bpasero** [comment]
We should take this to UX meeting.

**Comment by stevencl** [comment]
Even if there is no context menu available for a focusable UI panel I think we should still allow the user to tab into it to set focus. This lets them know and discover the different UI panels and components. For example, a screen reader might read out something about the empty panel.

**Comment by bpasero** [comment]
Yeah, also a good point. In fact every tree has a meaningful aria label assigned now.

**Comment by bpasero** [comment]
Closing.
